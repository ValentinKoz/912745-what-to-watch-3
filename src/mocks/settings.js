const genres = [`All genres`, `Adventure`, `Action`, `Comdey`, `Dramas`];

export {genres};
