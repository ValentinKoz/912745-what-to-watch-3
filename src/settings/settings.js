const MONTHS = [`January`, `February`, `March`, `April`, `May`, `June`, `July`, `August`, `September`, `October`, `November`, `December`];

const Path = {
  LOGIN: `/login`,
  FILMS: `/films`,
  MY_LIST: `/myList`,
  MAIN: `/`,
  REVIEW: `/review`,
  PLAYER: `/player`,
  PROMO: `/promo`,
  COMMENTS: `/comments`,
  FAVORITE: `/favorite`,
};

const Namespace = {DATA: `DATA`, STATE: `STATE`, USER: `USER`};

const DISPLAYED_ITEMS = 8;

const ADD_TO_SHOW = 8;

export {MONTHS, ADD_TO_SHOW, DISPLAYED_ITEMS, Namespace, Path};

export const adaptedObject = (item) => {
  return {
    id: `${item.id}`,
    name: item.name,
    poster: item.poster_image,
    preview: item.preview_image,
    backgroundImg: item.background_image,
    backgroundColor: item.background_color,
    video: item.video_link,
    previewVideo: item.preview_video_link,
    description: item.description,
    rating: item.rating,
    scoresCount: item.scores_count,
    director: item.director,
    starring: item.starring,
    runTime: item.run_time,
    genre: item.genre,
    released: item.released,
    isFavorite: item.is_favorite
  };
};

export const adaptedAuth = (item) => {
  return {
    id: `${item.id}`,
    email: item.email,
    name: item.name,
    avatarUrl: item[`avatar_url`],
  };
};

export const getDate = (propDate) => {
  const date = new Date(propDate);
  const month = MONTHS[date.getMonth()];
  const days = date.getDay();
  const year = date.getFullYear();
  return `${month} ${days}, ${year}`;
};

export const getRatingLevel = (rating) => {
  if (rating >= 0 && rating < 3) {
    return `Bad`;
  } else if (rating >= 3 && rating < 5) {
    return `Normal`;
  } else if (rating >= 5 && rating < 8) {
    return `Good`;
  } else if (rating >= 8 && rating < 10) {
    return `Vary good`;
  } else if (rating === 10) {
    return `Awesome`;
  }
  return ``;
};
