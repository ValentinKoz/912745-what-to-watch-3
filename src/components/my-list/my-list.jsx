import React, {PureComponent} from "react";
import PropTypes from "prop-types";
import {Link} from "react-router-dom";
import ListMovie from "./../list-movie/list-movie.jsx";
import {withActiveItem} from "../../hocs/with-active-item/with-active-item.js";
import {Namespace} from "../../settings/settings.js";
import {Operation} from "../../reducer/data/data.js";
import {connect} from "react-redux";

const ListMovieWrapped = withActiveItem(ListMovie);

class MyList extends PureComponent {

  componentDidMount() {
    const {loadFilms} = this.props;
    loadFilms();
  }

  render() {
    const {favoriteFilms, onCardClickHandle, authInfo} = this.props;
    return <div className="user-page">
      <header className="page-header user-page__head">
        <div className="logo">
          <Link to="/" className="logo__link">
            <span className="logo__letter logo__letter--1">W</span>
            <span className="logo__letter logo__letter--2">T</span>
            <span className="logo__letter logo__letter--3">W</span>
          </Link>
        </div>

        <h1 className="page-title user-page__title">My list</h1>

        <div className="user-block">
          <div className="user-block__avatar">
            <img src={`https://htmlacademy-react-3.appspot.com/${authInfo.avatarUrl}`} alt="User avatar" width="63" height="63" />
          </div>
        </div>
      </header>

      <section className="catalog">
        <h2 className="catalog__title visually-hidden">Catalog</h2>
        <ListMovieWrapped films={favoriteFilms || []} onCardClickHandle={onCardClickHandle} />
      </section>

      <footer className="page-footer">
        <div className="logo">
          <Link to="/" className="logo__link logo__link--light">
            <span className="logo__letter logo__letter--1">W</span>
            <span className="logo__letter logo__letter--2">T</span>
            <span className="logo__letter logo__letter--3">W</span>
          </Link>
        </div>

        <div className="copyright">
          <p>Â© 2019 What to watch Ltd.</p>
        </div>
      </footer>
    </div>;
  }
}

const mapStateToProps = (state) => ({
  favoriteFilms: state[Namespace.DATA].favoriteFilms,
  authInfo: state[Namespace.USER].authInfo,
});

const mapDispatchToProps = (dispatch) => ({
  loadFilms() {
    dispatch(Operation.loadFavorite());
  },
});

MyList.propTypes = {
  authInfo: PropTypes.shape({
    id: PropTypes.string,
    email: PropTypes.string,
    name: PropTypes.string,
    avatarUrl: PropTypes.string,
  }),
  favoriteFilms: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    poster: PropTypes.string.isRequired,
    preview: PropTypes.string.isRequired,
    backgroundImg: PropTypes.string.isRequired,
    backgroundColor: PropTypes.string.isRequired,
    video: PropTypes.string.isRequired,
    previewVideo: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    rating: PropTypes.number.isRequired,
    scoresCount: PropTypes.number.isRequired,
    director: PropTypes.string.isRequired,
    starring: PropTypes.arrayOf(PropTypes.string).isRequired,
    runTime: PropTypes.number.isRequired,
    genre: PropTypes.string.isRequired,
    released: PropTypes.number.isRequired,
    isFavorite: PropTypes.bool.isRequired,
  })),
  loadFilms: PropTypes.func.isRequired,
  onCardClickHandle: PropTypes.func.isRequired,
};

export {MyList};
export default connect(mapStateToProps, mapDispatchToProps)(MyList);
