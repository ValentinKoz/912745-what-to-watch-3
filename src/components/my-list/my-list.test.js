import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import {MyList} from "./my-list.jsx";
import {Namespace} from "./../../settings/settings.js";
import {BrowserRouter} from "react-router-dom";

const films = [{}, {}, {}];

const mockStore = configureStore([]);

const film = [{
  id: `1`,
  name: `Name`,
  poster: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Snatch.jpg`,
  preview: `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/snatch.jpg`,
  backgroundImg: `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/Snatch.jpg`,
  backgroundColor: `#FDFDFC`,
  video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
  previewVideo: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
  description: `text...`,
  rating: 8.0,
  scoresCount: 43241,
  director: `Guy Ritchie`,
  starring: [`Jason Statham`, `Brad Pitt`, `Benicio Del Toro`],
  runTime: 104,
  genre: `Comedy`,
  released: 2000,
  isFavorite: true
}];

const displayedItems = 8;

it(`PageMovie is renderer correctly`, () => {
  const store = mockStore({
    [Namespace.DATA]: {
      films,
      promo: films[0],
      favoriteFilms: film,
    },
    [Namespace.STATE]: {
      genre: `All genres`,
      displayedItems,
    },
    [Namespace.USER]: {
      authorizationStatus: `AUTH`,
      authInfo: {
        "id": `1`,
        "email": `Oliver.conner@gmail.com`,
        "name": `Oliver.conner`,
        "avatarUrl": `img/1.png`
      }
    }
  });
  const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <MyList onCardClickHandle={() => {}} favoriteFilms={film} loadFilms={()=>{}} authInfo={{
            "id": `1`,
            "email": `Oliver.conner@gmail.com`,
            "name": `Oliver.conner`,
            "avatarUrl": `img/1.png`
          }}/>
        </BrowserRouter>
      </Provider>, {
        createNodeMock: () => {
          return {};
        }
      }).toJSON();

  expect(tree).toMatchSnapshot();
});
