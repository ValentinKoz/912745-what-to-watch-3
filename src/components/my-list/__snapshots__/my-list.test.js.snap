// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageMovie is renderer correctly 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      My list
    </h1>
    <div
      className="user-block"
    >
      <div
        className="user-block__avatar"
      >
        <img
          alt="User avatar"
          height="63"
          src="https://htmlacademy-react-3.appspot.com/img/1.png"
          width="63"
        />
      </div>
    </div>
  </header>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <div
      className="catalog__movies-list"
    >
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="small-movie-card__image"
        >
          <video
            height="175"
            poster="https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/snatch.jpg"
            src="http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4"
            width="280"
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="#"
            onClick={[Function]}
          >
            Name
          </a>
        </h3>
      </article>
       
    </div>
  </section>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        Â© 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
